### PR TITLE
feat: Reproduce Figure 2 from arXiv:2602.20299 (SAT-MPS Entanglement Barriers)

### PR DESCRIPTION
**Pull request by QingyunQian**
## Summary
This pull request adds a reproduction of Figure 2 from the paper "Entanglement Barriers from Computational Complexity: Matrix-Product-State Approach to Satisfiability" (arXiv:2602.20299).
It implements the 3-SAT instance generator and the imaginary-time evolution entanglement analysis using the `jax` backend in TensorCircuit-NG. 
Default parameters scaled down to n=4-15 (Panel A), n=8-10 (Panel B), 200 instances for laptop-friendly runtimes; full paper settings available via CLI args.

Code has been checked with `black` and `pylint`.

## Checklist

- [x] Single `main.py` with `meta.yaml` and `outputs/result.png`
- [x] `black` formatted
- [x] `pylint` score 10/10
- [x] Runs successfully with `python main.py`
- [x] No large binary files

### CODE REVIEW

**Review by refraction-ray — COMMENTED** [review]
is gradient descent for the peak position possible? the current implementation is too complicated for the peak locating, for me, the original scipy minimize is even better than the current implementation for a demonstration script where the performance and better practice are both very important as it can guide the patterns of further development via ai agents. The current impl for peak locating is somehow too specific and a bit anti-pattern for tcng to me.

File: examples/reproduce_papers/2026_sat_mps_entanglement/main.py [hunk 1]
```diff
@@ -35,7 +31,7 @@
 }
 
 K = tc.set_backend("jax")
-tc.set_dtype("complex128")
+tc.set_dtype("float64")
```
[the remark this answers is not in the document]
Comment (QingyunQian) [reply examples/reproduce_papers/2026_sat_mps_entanglement/main.py]: thx a lot, I checked the JAX-side alternative as well, but jax.scipy.optimize.minimize currently only supports BFGS( this API only realize part of scipy solver, there is no bounded scalar optimizer solver here, i.e. scipy.optimize.minimize_scalar(method="bounded") ), and also assumes a differentiation friendly objective. I tried it and led to mathematical failures and performance degradation for this specific physical system. (Depending on the instance and τ, density matrix here may develop very small or nearly degenerate eigenvalues, which can make gradient-based optimization through spectral decompositions numerically fragile)

[the remark this answers is not in the document]
Comment (refraction-ray) [reply examples/reproduce_papers/2026_sat_mps_entanglement/main.py]: I guess the reason that you function has no gradient support and performance regression is because you use compute_uv=False for entropy, if ad is required, uv have to be recomputed anyway.
